src/OrleansAzureUtils/Storage/AzureTableDataManager.cs

```diff
@@ -467,57 +437,93 @@ public AzureTableDataManager(string tableName, string storageConnectionString, L
+            }
+            finally
+            {
+                CheckAlertSlowAccess(startTime, operation);
+            }
+        }
+        /// <summary>
+        /// Read data entries and their corresponding eTags from the Azure table.
+        /// </summary>
+        /// <param name="filter">Predicate function to use for querying the table and filtering the results.</param>
+        /// <returns>Enumeration of entries in the table which match the query condition.</returns>
+        private async Task<IEnumerable<Tuple<T, string>>> ReadTableEntriesAndEtagsWithTableQueryAsync(TableQuery<T> cloudTableQuery)
+        {
+            try
+            {
+                Func<Task<List<T>>> executeQueryHandleContinuations = async () =>
                 {
-                    // Out of retries...
-                    var errorMsg = $"Failed to read Azure storage table {TableName}: {exc.Message}";
-                    if (!AzureStorageUtils.TableStorageDataNotFound(exc))
+                    TableQuerySegment<T> querySegment = null;
+                    var list = new List<T>();
+                        
+                    while (querySegment == null || querySegment.ContinuationToken != null)
                     {
-                        Logger.Warn(ErrorCode.AzureTable_09, errorMsg, exc);
+                        querySegment = await tableReference.ExecuteQuerySegmentedAsync(cloudTableQuery, querySegment?.ContinuationToken);
+                        list.AddRange(querySegment);
                     }
-                    throw new OrleansException(errorMsg, exc);
-                }
+                    return list;
+                };
+
+                IBackoffProvider backoff = new FixedBackoff(AzureTableDefaultPolicies.PauseBetweenTableOperationRetries);
+
+                List<T> results = await AsyncExecutorWithRetries.ExecuteWithRetries(
+                    counter => executeQueryHandleContinuations(),
+                    AzureTableDefaultPolicies.MaxTableOperationRetries,
+                    (exc, counter) => AzureStorageUtils.AnalyzeReadException(exc.GetBaseException(), counter, TableName, Logger),
+                    AzureTableDefaultPolicies.TableOperationTimeout,
+                    backoff);
+
+                // Data was read successfully if we got to here                    
+                return results.Select(i => Tuple.Create(i, i.ETag)).ToList();
+
             }
-            finally
+            catch (Exception exc)
             {
-                CheckAlertSlowAccess(startTime, operation);
+                // Out of retries...
+                var errorMsg = $"Failed to read Azure storage table {TableName}: {exc.Message}";
+                if (!AzureStorageUtils.TableStorageDataNotFound(exc))
+                {
+                    Logger.Warn(ErrorCode.AzureTable_09, errorMsg, exc);
+                }
+                throw new OrleansException(errorMsg, exc);
             }
```

Comment:
it was moved up to ReadTableEntriesAndEtagsWithFilterString. But since we are merging these two back together, this doesn't matter anymore. :) 

In one word, it is there.